pysynchronizer/cmd.py

```diff
@@ -0,0 +1,207 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+from cmd import Cmd
+from os.path import split
+
+from .errors import ConnectionError
+from .objects import XenMgr, VM
+from .utils import is_valid_uuid, column_print, path_to_uuid, uuid_to_path
+
+class BaseCmd(Cmd):
+    def __init__(self):
+        Cmd.__init__(self)
+
+    def run(self, cmd_str=""):
+        if cmd_str == "":
+            self.cmdloop()
+        else:
+            self.onecmd(cmd_str)
+
+    def do_exit(self, args):
+        return -1
+
+class SyncCmd(BaseCmd):
+    def __init__(self, cmd_str=""):
+        BaseCmd.__init__(self)
+        self.prompt = "sync> "
+
+    def do_xenmgr(self, arg_str):
+        try:
+            XenMgrCmd().run(arg_str)
+        except CommuincationError as err:
+            print("Connection failed:\n\t%s\n" % err)
+        except Exception as err:
+            print("Unexpected exception:\n\t%s\n" % err)
+
+    def do_vm(self, arg_str):
+        args = arg_str.split()
+
+        try:
+            if args:
+                vc = VmCmd(args[0])
+                arg_str = " ".join(args[1:])
+            else:
+                vc = VmCmd()
+                arg_str = ""
+
+            vc.run(arg_str)
+        except CommuincationError as err:
+            print("Connection failed:\n\t%s\n" % err)
+        except Exception as err:
+            print("Unexpected exception:\n\t%s\n" % err)
+
+class XenMgrCmd(BaseCmd):
+    def __init__(self):
+        BaseCmd.__init__(self)
+
+        self.prompt = "xemmgr> "
+        self.xenmgr = XenMgr()
+
+    def do_list_vm(self, arg_str):
+        rows = [ [
+            "Name",
+            "UUID",
+        ] ]
+
+        vm_list = self.xenmgr.vm_by_name()
+        for name in vm_list:
+            rows.append([name, vm_list[name]])
+
+        column_print(rows)
+
+    def do_create_vm(self ,arg_str):
+        pass
+
+    def do_delete_vm(self, arg_str):
+        ref = arg_str.split()[0]
+        vm_uuid = self.xenmgr.find_vm(ref)
+
+        if vm_uuid:
+            vm = VM(vm_uuid)
+            vm.delete()
+        else:
+            print("unable to find vm: %s" % ref)
+
+    def do_upgrade(self, args):
+        if not self.xenmgr.upgrade(args.split()[0]):
+            print("upgrade failed")
+
+class VmCmd(BaseCmd):
+    def __init__(self, arg_str):
+        BaseCmd.__init__(self)
+
+        self.prompt = "vm> "
+
+        if len(arg_str) > 0:
+            self.select_vm(arg_str.split()[0])
+
+    def select_vm(self, vm_ref):
+        vm_uuid = XenMgr().find_vm(vm_ref)
+        if vm_uuid:
+            self.vm = VM(vm_uuid)
+            self.name = self.vm.get_property("name")
+            self.prompt = "vm[%s]> " % self.name
```

Comment:
self.name isn't used otherwise, so just set the prompt directly?

Reply:
Yah, this morphed a few times, so use got dropped. I can drop here as well.